### PR TITLE
Created Cart&Wishlist context

### LINKED ISSUE #9: Create Cart&Wishlist context
## Todo
- Create Cart&Wishlist context
- Add Cart & wishlist functionalities in the product card
- Create Cart page and wishlist page

### PR DESCRIPTION
**Pull request by kjais1720**
PR for issue #9 
## Overview
- Created Cart and wishlist context
- updated loader style

### CODE REVIEW

File: src/contexts/cart&wishlist/CartWishlistProvider.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  });
+  const { serverResponse, isLoading } = useAxios(
+    cartWishlistState.apiUrl,
+    cartWishlistState.apiMethod,
+    cartWishlistState.postData,
+    encodedToken
+    );
+  useEffect(() => {
+    console.log(serverResponse)
+    // To update the cart and wishlist state everytime the app gets new data from the server
```
Comment (raman1999) [inline src/contexts/cart&wishlist/CartWishlistProvider.jsx:41]: remove this console

File: src/contexts/cart&wishlist/CartWishlistProvider.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  useEffect(() => {
+    let timeoutId;
+    if (!isUserAuthenticated) {
+      cartWishlistDispatch({ type: "clearCart" })
+      setCartItems([]);
+      setwishlistItems([]);
+    }
+    else {
+      console.log({isUserAuthenticated})
+      cartWishlistDispatch({ type: "getCartList" });
```
Comment (raman1999) [inline src/contexts/cart&wishlist/CartWishlistProvider.jsx:59]: here also

**Review by raman1999 — APPROVED** [review]
Very clean code written, and I like the way u are  using reducer  and axios  together for updating, good logical thinking done

**Review by rahulcs754 — APPROVED** [review]
like the way to call api and access data.
Code looks clean

File: src/contexts/cart&wishlist/CartWishlistProvider.jsx (hunk 2 above)
Comment (kjais1720) [reply src/contexts/cart&wishlist/CartWishlistProvider.jsx:59]: Thanks for pointing it out, I have removed it in the latest commit

File: src/contexts/cart&wishlist/CartWishlistProvider.jsx (hunk 1 above)
Comment (kjais1720) [reply src/contexts/cart&wishlist/CartWishlistProvider.jsx:41]: Thanks for pointing it out, I have removed it in the latest commit

**Pull request merged**